samples/03.speech/h.select-audio-input-device/comprehensive.html

```diff
@@ -0,0 +1,246 @@
+<!DOCTYPE html>
```

Comment:
I'm not convinced that this sample works as a single `.html` page. It's very complicated and long, and it's using React concepts that may be unfamiliar to the user. At the very least, I would like to see the styling moved to a `.css` page. I know this doesn't match other samples, but the styling is purely aesthetic in this case.

Reply:
I think it's time that we move functions that are commonly used in most samples to a separate `.js` file.  These file lengths are unwieldy.